paddle/fluid/operators/controlflow/conditional_block_op.cc

```diff
@@ -161,23 +248,56 @@ class ConditionalBlockGradOp : public ConditionalOp {
           platform::errors::InvalidArgument(
               "Expect Scope variable contains at least 1 scope, but got: %d",
               scopes.size()));
-      framework::Scope &cur_scope = *scopes[0];
+      framework::Scope &cur_scope = *(scopes[0]);
 
       auto *block = Attr<framework::BlockDesc *>("sub_block");
-
       VLOG(3) << "Conditional Grad block.idx = " << block->ID()
               << ", scope = " << &cur_scope;
-      if (!exec || !platform::is_same_place(exec->GetPlace(), dev_place)) {
-        auto &pdesc = *block->Program();
-        exec.reset(new Executor(dev_place));
-        if (FLAGS_use_mkldnn) exec->EnableMKLDNN(pdesc);
-        ctx = exec->Prepare(pdesc, block->ID(), inside_grads, false);
+
+      if (FLAGS_control_flow_use_new_executor) {
+        std::set<std::string> skip_gc_vars(inside_grads.begin(),
+                                           inside_grads.end());
+        bool old_flag_new_executor_use_loacl_scope =
+            FLAGS_new_executor_use_local_scope;
+        FLAGS_new_executor_use_local_scope = false;
```

Comment:
Because there's no need to create a new scope when executing `ConditionalBlockOp`. I will change the use of flag to initialization parameters of `InterpreterCore`.

Reply:
Done